### PR TITLE
Max Enrollments Number

### PR DESCRIPTION
**Pull request by sameer-ah**
Fix: #142
-If 'Allow enrollment to multiple groups' is selected, there is an extra field to provide max enrollments number. Use default value 0 if there is no limit.

### CODE REVIEW

File: view.php [hunk 1]
```diff
@@ -33,6 +33,7 @@
 $action     = optional_param('action', '', PARAM_ALPHA);
 $userids    = optional_param_array('userid', array(), PARAM_INT); // array of attempt ids for delete action
 $notify     = optional_param('notify', '', PARAM_ALPHA);
+$maxenrollments     = optional_param('maxenrollments', '', PARAM_INT);
```
Comment (ndunand) [inline view.php:36]: Do we need this optional_param here? More simply, we could simply use `$choicegroup->maxenrollments` throughout the code.

File: view.php [hunk 2]
```diff
@@ -102,6 +103,21 @@
...
+        if($maxenrollments > 0){
+            for ($i = 0; $i < $number_of_groups; $i++) {
+                $answer_value = optional_param('answer_' . $i, '', PARAM_INT);
+                if ($answer_value != '') {
+                    $enrollmentscount++;
+                }
+            }
+            if ($enrollmentscount > $maxenrollments) {
+                redirect(new moodle_url('/mod/choicegroup/view.php',
+                    array('id' => $cm->id, 'notify' => 'mustchoosemax', 'sesskey' => sesskey(), 'maxenrollments' => $maxenrollments)));
```
Comment (ndunand) [inline view.php:118]: ... so we can remove the redundant `maxenrollments ` redirect param here.

**Review by ndunand — CHANGES_REQUESTED** [review]
Thanks for sharing your work on this!

Just 2 small comments that I'd ask you to consider.

**Comment by sameer-ah** [comment]
Thanks for the review feedback. I have made suggested changes.

**Comment by ndunand** [comment]
Thanks @sameer-ah for providing this feature! This will be released asap.

**Pull request merged**